April news (#755)
ropensci/roweb3: content/blog/2024-04-22-news-april-2024/index.Rmd

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

### Coworking

Read [all about coworking](/blog/2023/06/21/coworking/)!

Join us for social coworking & office hours monthly on first Tuesdays! 
Hosted by Steffi LaZerte and various community hosts. 
Everyone welcome. 
No RSVP needed. 
Consult our [Events](/events) page to find your local time and how to join.

- [Tuesday, May 7th, 09:00 Americas Pacific (16:00 UTC)](/events/coworking-2024-05), Meet rOpenSci's new Executive Director! With cohost [Noam Ross](/author/noam-ross) and [Steffi LaZerte](/author/steffi-lazerte).
    - Learn more about rOpenSci!
    - Explore rOpenSci [packages](/packages) and [contributing guide](https://contributing.ropensci.org/)
    - Ask questions or chat about rOpensci with Noam and other attendees.
    
And remember, you can always cowork independently on work related to R, work on packages that tend to be neglected, or work on what ever you need to get done!

## Software :package:

### GSOD funding placeholder

TODO MAËLLE

> ### Social media placeholder

Reviewer: ```suggestion
### Learn about a different package every day by following our social media campaigns 
```

TODO YANI

> ### Comm calls ideas placeholder

Reviewer: ```suggestion
### Help us put together the rOpenSci Community Call calendar for the year!
```

> TODO YANI

Reviewer: ```suggestion
We are organizing the Community Calls for this year. Could you help us decide which topics we should cover? 

We hold [Community Calls](/commcalls/) to share knowledge that is relevant to our community and consistent with [our vision and mission](/about/). These are free and open for anyone to attend and provide opportunities for us to connect with rOpenSci community members around the world.
We’ve [opened an issue in this repository for each topic we’re considering](https://github.com/ropensci-org/community-calls/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc). We’d like your input and “votes” on these, and your suggestions for other topics we haven’t thought of.

We would love to hear your ideas. 🙏 
```

---
title: 'rOpenSci News Digest, April 2024'
author: 'The rOpenSci Team'
date: '2024-04-22'
tags:
  - newsletter

> slug: ropensci-news-march-2024

Reviewer 2: ```suggestion
slug: ropensci-news-april-2024
```

}

nice_string <- function(...) {
  if (length(...) == 2) {
    return(paste(c(...), collapse = " and "))
  }
  glue::glue_collapse(..., sep = ", ", last = ", and ")
}
```
<!-- Before sending DELETE THE INDEX_CACHE and re-knit! -->

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

### Coworking

Read [all about coworking](/blog/2023/06/21/coworking/)!

Join us for social coworking & office hours monthly on first Tuesdays! 
Hosted by Steffi LaZerte and various community hosts. 
Everyone welcome. 
No RSVP needed. 
Consult our [Events](/events) page to find your local time and how to join.

- [Tuesday, May 7th, 09:00 Americas Pacific (16:00 UTC)](/events/coworking-2024-05), Meet rOpenSci's new Executive Director! With cohost [Noam Ross](/author/noam-ross) and [Steffi LaZerte](/author/steffi-lazerte).
    - Learn more about rOpenSci!
    - Explore rOpenSci [packages](/packages) and [contributing guide](https://contributing.ropensci.org/)
    - Ask questions or chat about rOpensci with Noam and other attendees.

>     

Reviewer 2: I think Emi has agreed, but we may have had a time mix up and I don't know that we'll have it resolved by Monday. 
```suggestion
    
- Tuesday, June 4th, 09:00 Australia Western (01:00 UTC), Theme TBD with cohost TBD and [Steffi LaZerte](/author/steffi-lazerte).
```

And remember, you can always cowork independently on work related to R, work on packages that tend to be neglected, or work on what ever you need to get done!

## Software :package:

> ### R-Universe Documentation Gets a Boost from Google Season of Docs

Reviewer 2: ```suggestion
### R-Universe documentation gets a boost from Google Season of Docs
```

We are excited and grateful to [announce](/blog/2024/04/12/gsod-announcement/) that [R-Universe](https://r-universe.dev/) has been awarded a [Google Season of Docs](https://developers.google.com/season-of-docs) grant. R-Universe is rOpenSci’s platform for testing, building, distributing, and discovering R packages, led by [Jeroen Ooms](/author/jeroen-ooms/).

With the support of Google, we aim to update and centralize our documentation, showcasing how users and developers can get the most out of R-Universe. We will update the documentation to catch up with the rapid feature development of the past two years, including consolidating our many tech notes and READMEs into an easily discoverable site. We will build the site with Quarto to make it straightforward for R community members to contribute.

The work will be led by [Maëlle Salmon](/author/maëlle-salmon/).

### Learn about a different package every day by following our social media campaigns 

> The goal of a community is to collaborate, share knowledge and build learnings.  That's why we want to show you the more than 300 packages in our federated community of open-source software developers. 

Reviewer 2: I'm not sure about the word "federated" here... I don't think it's wrong (technically), but it definitely makes me think of Mastodon...
```suggestion
The goal of a community is to collaborate, share knowledge and learn together.  That's why we want to share with you the more than 300 packages in our federated community of open-source software developers. 
```

> Since December 2023, we have run the social media campaigns [“A Package a Day](https://hachyderm.io/@rOpenSci/tagged/APackageADay) on [Mastodon](https://hachyderm.io/@rOpenSci), in which we feature one package each day. On [LinkedIn](https://www.linkedin.com/company/ropensci/), we publish our [Package Weekly Digest](https://www.linkedin.com/feed/hashtag/?keywords=packageweeklydigest&highlightedUpdateUrns=urn%3Ali%3Aactivity%3A7170464303186108416) every Monday with a list of five packages that belong to the same category.

Reviewer 2: ```suggestion
Since December 2023, we have been running social media campaigns on [Mastodon](https://hachyderm.io/@rOpenSci) and [LinkedIn](https://www.linkedin.com/company/ropensci/). On Mastodon, we have [“A Package a Day](https://hachyderm.io/@rOpenSci/tagged/APackageADay), in which we feature one package each day. On LinkedIn, we publish our [Package Weekly Digest](https://www.linkedin.com/feed/hashtag/?keywords=packageweeklydigest&highlightedUpdateUrns=urn%3Ali%3Aactivity%3A7170464303186108416) every Monday with a list of five packages that belong to the same category.
```

> You can follow us and the hashtags #APackageADay and #PackageWeeklyDigest to stay informed about R packages and learn about them.

Reviewer 2: ```suggestion
You can follow us and the hashtags #APackageADay and #PackageWeeklyDigest to stay informed about rOpenSci R packages and learn about them.
```

* [historydata](https://docs.ropensci.org/historydata/), datasets for historians. [Issue for volunteering](https://github.com/ropensci/historydata/issues/23).

* [textreuse](https://docs.ropensci.org/textreuse/), detect text reuse and document similarity. [Issue for volunteering](https://github.com/ropensci/textreuse/issues/97).

* [USAboundaries](https://docs.ropensci.org/USAboundaries/) (and USAboundariesdata), historical and contemporary boundaries of the United States of America . [Issue for volunteering](https://github.com/ropensci/USAboundaries/issues/50).

### Calls for contributions

* [Help make qualtRics better! Code contributions wanted](/blog/2024/03/01/qualtrics-call-help/)

* [Help make assertr better! Come close issues](/blog/2024/02/27/assertr-call-help/)

* [Help users and developers of targets by answering questions!](/blog/2024/02/29/targets-call-help/)

* [Help make waywiser better! User requests wanted](/blog/2024/02/19/waywiser-call-help/)

Also refer to our [help wanted page](/help-wanted/) -- before opening a PR, we recommend asking in the issue whether help is still needed.

## Package development corner

Some useful tips for R package developers. :eyes:

> ### R-hub version 2!

Reviewer 2: 🎉

Gábor Csárdi announced the [second version of R-hub](https://blog.r-hub.io/2024/04/11/rhub2/), for checking your R package, either on GitHub Actions or in an ad-hoc basis, using the rhub2 package.

### Patterns and anti-patterns of data analysis reuse

> Miles McBain wrote an [insightful blog](https://www.milesmcbain.com/posts/data-analysis-reuse/) post advocating for creating an universe of related packages to support data analyses at your organization, contrasting this approach with others such as copy-pasting or having a single gigantic package.

Reviewer 2: ```suggestion
Miles McBain wrote an [insightful blog](https://www.milesmcbain.com/posts/data-analysis-reuse/) post advocating for creating a universe of related packages to support data analyses at your organization, contrasting this approach with other approaches such as copy-pasting or having a single gigantic package.
```

> ### Install several R versions at once with rig

Reviewer 2: 😮

As reminded by [Jenny Bryan on Mastodon](https://mastodon.social/@jennybryan@fosstodon.org/112176595542760673), rig, the [R Installation Manager](https://github.com/r-lib/rig#readme) maintained by Gábor Csárdi, is a fantastic tool for being able to experiment locally with different R versions, especially when developing or debugging your R packages.

### Update on mocking for testing R packages

Mocking is the "art of replacing a function with whatever fake we need for testing", a nice tool to add to your toolbox. 
Read the [R-hub blog post](https://blog.r-hub.io/2024/03/21/mocking-new-take/).

### Be careful when using git clean

Athanasia Mo Mowinckel shared her recent [Git mishap](https://drmowinckels.io/blog/2024/2024/git-clean-woes/) in a helpful post that also includes tips and resources.

### Notes on refactoring an API client

> [Scott Chamberlain](https://ropensci.org/author/scott-chamberlain/) published some [notes about refactoring an API client](https://recology.info/2024/03/refactoring-notes/), including some arguments in favor of "separation of concerns and code".

Reviewer 2: ```suggestion
[Scott Chamberlain](/author/scott-chamberlain/) published some [notes about refactoring an API client](https://recology.info/2024/03/refactoring-notes/), including some arguments in favor of "separation of concerns and code".
```
